### PR TITLE
Add a cmake variable to choose if the pyhton wrapper will be compile or not

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
Allow the user to not compile the python binding if unnecessary

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -159,6 +160,7 @@ SET(${PROJECT_NAME}_ALGORITHM_HEADERS
   algorithm/compute-all-terms.hpp
   )
 
+IF(${BUILD_PYTHON_INTERFACE} STREQUAL "ON")
```
Comment (jcarpent) [inline CMakeLists.txt]: Minor remark. For future, here you can directly  use IF(BUILD_PYTHON_INTERFACE).

**Comment by jcarpent** [comment]
Related to #168. 
@MaximilienNaveau Thanks for this contribution.

File: CMakeLists.txt (hunk 1 above)
Comment (MaximilienNaveau) [inline CMakeLists.txt]: In fact I tried and the cmake failed as the value of  `IF(BUILD_PYTHON_INTERFACE)` is a sstring with value 'ON'
Maybe I missed something. Though it should work properly.

Comment (jcarpent) [inline CMakeLists.txt]: But BUILD_PYTHON_INTERFACE is not a string, that's the point.

**Pull request merged**